src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -11,33 +15,51 @@
 
 public class ShooterSubsystem extends SubsystemBase {
 
-  private final CANSparkMax shooterTopMotor;
-  private final CANSparkMax shooterBottomMotor;
+  private final CANSparkMaxSim shooterTopMotor;
+  private final CANSparkMaxSim shooterBottomMotor;
+
+  private final RelativeEncoder shooterTopMotorEncoder;
+  private final RelativeEncoder shooterBottomMotorEncoder;
+
+  private ShuffleboardTab ShooterTab = Shuffleboard.getTab("Shooter Tab");
 
   private DebugEntry<Double> topMotorSpeedEntry;
-  private DebugEntry<Double> bottomMotorSpeedEntry;
   private DebugEntry<Double> topMotorTargetSpeedEntry;
+  private DebugEntry<Double> topMotorTemperatureEntry;
+
+  private DebugEntry<Double> bottomMotorSpeedEntry;
   private DebugEntry<Double> bottomMotorTargetSpeedEntry;
+  private DebugEntry<Double> bottomMotorTemperatureEntry;
+
+  private DebugEntry<Boolean> shooterReadyEntry;
```

Comment:
None of these are instantiated. This code will crash as soon as a .log() method is hit